<?xml version="1.0" encoding="UTF-8"?>
<spring:beans xmlns="http://www.citrusframework.org/schema/testcase" 
              xmlns:spring="http://www.springframework.org/schema/beans" 
              xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" 
              xsi:schemaLocation="http://www.springframework.org/schema/beans 
                                  http://www.springframework.org/schema/beans/spring-beans-2.5.xsd 
                                  http://www.citrusframework.org/schema/testcase 
                                  http://www.citrusframework.org/schema/testcase/citrus-testcase.xsd">
    <testcase name="AssertITest">
        <meta-info>
            <author>Christoph Deppisch</author>
            <creationdate>2008-06-16</creationdate>
            <status>FINAL</status>
            <last-updated-by>Christoph Deppisch</last-updated-by>
            <last-updated-on>2008-06-16T00:00:00</last-updated-on>
        </meta-info>
        
        <description>Asserting excpetions</description>
        
        <actions>
            <assert exception="com.consol.citrus.exceptions.CitrusRuntimeException">
            	<fail message="Fail once"/>
            </assert>
            <assert exception="com.consol.citrus.exceptions.CitrusRuntimeException"
                    message="Fail again">
            	<fail message="Fail again"/>
            </assert>
            
            <assert exception="com.consol.citrus.exceptions.ValidationException">
              <assert exception="java.io.IOException">
                  <fail message="Fail another time"/>
              </assert>
            </assert>
            
            <assert exception="com.consol.citrus.exceptions.ValidationException">
              <assert exception="com.consol.citrus.exceptions.CitrusRuntimeException"
                      message="Fail again">
                  <fail message="Fail with nice error message"/>
              </assert>
            </assert>
            
            <assert exception="com.consol.citrus.exceptions.ValidationException">
              <assert exception="com.consol.citrus.exceptions.CitrusRuntimeException">
                  <echo>
                    <message>Nothing fails here</message> 
                  </echo>
              </assert>
            </assert>
            
            <assert exception="com.consol.citrus.exceptions.ValidationException">
              <assert exception="com.consol.citrus.exceptions.CitrusRuntimeException"
                      message="Must be failing">
                  <echo>
                    <message>Nothing fails here either</message> 
                  </echo>
              </assert>
            </assert>
        </actions>
    </testcase>
</spring:beans>
